R/sf.R

```diff
@@ -476,9 +476,20 @@ CoordSf <- ggproto("CoordSf", CoordCartesian,
     if (!is.null(graticule$plot12))
       graticule$degree_label[!graticule$plot12] <- NA
 
-    # parse labels into expressions if required
-    if (any(grepl("degree", graticule$degree_label)))
-      graticule$degree_label <- lapply(graticule$degree_label, function(x) parse(text = x)[[1]])
+    # Convert the string 'degree' to the degree symbol
+    parse_ids <- grepl("\\bdegree\\b", graticule$degree_label)
+    if (any(parse_ids)) {
+      graticule$degree_label <- Map(
+        function(parse_id, label) {
+          if (parse_id) {
+            parse(text = label)[[1]]
+          } else {
+            as.expression(label)[[1]]
```

Comment:
Why are we turning `label` into an expression object and then immediately extracting it out of the expression object?

Reply:
Because we need the symbol itself, not `expression(symbol)`. This is directly copied from the current code.

Reply:
One problem with this code chunk as a whole, is that when I read it I can't easily tell what type of object `graticule$degree_label` is supposed to be. It's obviously a list, and some of the components are expressions (not [expression objects](https://adv-r.hadley.nz/expressions.html#naming-conventions)), but what are the others supposed to be? 

You could fix that with a comment, but it would be even better to use functions where it's obvious what the output type is.

Reply:
If we need a symbol, and we have a character vector of length one, then the function to use is `as.symbol()`

Reply:
Yes, correct. See also my other comment, below.